irony-completion.el

```diff
@@ -230,11 +230,17 @@ that can be validly accessed are deemed not-accessible."
   (nth 7 candidate))
 
 (defun irony-completion--filter-candidates (candidates)
-  (cl-remove-if-not
-   (lambda (candidate)
-     (memq (irony-completion-availability candidate)
-           irony-completion-availability-filter))
-   candidates))
+  (let (unique-candidates)
```

Comment:
I believe `(cons 1 (cons 2 (cons 3 nil)))` is the same as `(list 1 2 3)`.

I think the first `when` is useless now, as it is no longer the final statement.

Maybe you should use `(and (memq ...) (push unique-key ...)`.

Reply:
To clarify, I meant to rewrite the code like this:

```elisp
  (and (memq (irony-completion-availability candidate)
		   irony-completion-availability-filter))
	 (let ((unique-key (list (irony-completion-typed-text candidate)
                                 (irony-completion-annotation candidate)
                                 (irony-completion-type candidate))))
	   (and (not (member unique-key unique-candidates))
                (push unique-key unique-candidates)))))
```

2 `when`, makes the first useless:

```elisp
(defun (a b)
  (when a t)
  (when b t))
```

Here, whatever the value of `a`, `t` will be returned only if `b` is non-nil.

Reply:
Hello, just got back from work.

Yes, that makes sense.